packages/venia-concept/src/index.js

```diff
@@ -16,10 +17,28 @@ import app from '@magento/peregrine/lib/store/actions/app';
 import App, { AppContextProvider } from '@magento/venia-ui/lib/components/App';
 
 import { registerSW } from './registerSW';
+import {
+    toReactIntl,
+    fromReactIntl
+} from '@magento/venia-ui/lib/util/formatLocale';
 
 const { BrowserPersistence } = Util;
 const apiBase = new URL('/graphql', location.origin).toString();
 
+const language = toReactIntl(STORE_VIEW_LOCALE);
+let messages;
+
+try {
+    messages = require(`@magento/venia-ui/lib/i18n/${fromReactIntl(
+        language
+    )}.json`);
+} catch (err) {
+    console.warn(
+        `Unable to load translation file. Loading 'venia-ui/lib/i18n/en_US.json' instead. \n${err}`
+    );
+    messages = require(`@magento/venia-ui/lib/i18n/en_US.json`);
+}
```

Comment:
I believe we should use [dynamic import](https://webpack.js.org/api/module-methods/#dynamic-expressions-in-import) statements in the browser, rather than `require` statements. Note that these are asynchronous and that we can do some cool things with them, such as preloading.

Since I'm recommending that we switch from a synchronous (blocking) require to an async (non-blocking) one, that means we'll have to re-render `IntlProvider` when the call completes and we have real `messages` to give it. As a result, we should probably move this dynamic import into a component that wraps `IntlProvider`.

Finally, we should do as much of this as possible in `venia-ui` rather than `venia-concept`, I think. Making a component would be a good way to do that.

```es6
// shouldn't even have to provide a default, right?
// https://formatjs.io/docs/react-intl/components#defaultlocale-and-defaultformats
const locale = fromReactIntl(language)

const LocaleProvider = props => {
  const { children } = props
  const [messages, setMessages] = useState(null)

  // try doing the dynamic import as an effect
  // and set the result in state, so we can render before & after
  useEffect(
    () => {
      import(`@magento/venia-ui/lib/i18n/${locale}.json`)
        .then(data => { setMessages(data) })
        .catch(error => { console.error("Unable to load translation file.") })
    },
    [setMessages]
  )

  return (
    <IntlProvider
      key={language}
      {...props}
      defaultLocale="en-US"
      locale={language}
      messages={messages}
    />
  )
}